src/tools/tools/hash.fs

```diff
@@ -0,0 +1,173 @@
+﻿namespace Prajna.Tools
+
+#nowarn "9"
+    
+open System
+open System.Runtime.InteropServices
+open System.Security.Cryptography
+
+open FSharp.NativeInterop
+
+module Hash =
```

Comment:
"internal",  I don't assume we want to expose it to Prajna users as public APIs.